Assets/Scripts/CombatZone.cs

```diff
@@ -33,50 +38,97 @@ public void Start()
 
     }
 
+    public void Update()
+    {
+        if (!Timer.IsInProgress())
+            return;
+
+        if (CombatStarted)
+            DisablePlayerMovement();
+
+        if (!Timer.IsFinished())
+            SetCombatPositionsUpdate();
+    }
+
     void OnTriggerEnter(Collider other)
     {
         if (CombatStarted)
             return;
 
         if (other.gameObject.CompareTag("Player"))
         {
+            Timer.StartTimer(MovementTime);
+            StartCoroutine(StartingCombat());
             InitiateCombat();
         }
     }
 
     private void InitiateCombat()
     {
         CombatStarted = true;
+        InitalPositionsEnemies = new Vector3[Enemies.Length];
+        InitalPositionsPlayers = new Vector3[Players.Length];
 
-        SetCombatPositions();
+        for (int i = 0; i < InitalPositionsEnemies.Length; i++)
+        {
+            InitalPositionsEnemies[i] = Enemies[i].transform.position;
+            Enemies[i].GetComponent<Combatant>().EnterCombat();
+        }
 
-        CombatSystem.StartCombat(this.gameObject, Players, Enemies);
+        for (int i = 0; i < InitalPositionsPlayers.Length; i++)
+        {
+            Players[i].GetComponent<Combatant>().EnterCombat();
+            InitalPositionsPlayers[i] = Players[i].transform.position;
+        }
     }
 
-    private void SetCombatPositions()
+    private void DisablePlayerMovement()
+    {
+        if (Timer.IsFinished())
+            return;
+
+
+        for (int i = 0; i < PlayerPositions.Length; i++)
+        {
+            GameObject player = Players[i];
+            player.GetComponent<Player>().DisableMovement();
+            player.SetActive(false);
```

Comment:
these two lines are....? Not needed at all hopefully?

Reply:
I think they're needed cause we need to disable movement during combat.